lib/metasploit/framework/login_scanner/snmp.rb

```diff
@@ -161,6 +161,8 @@ def scan!
 
           shutdown_socket
           nil
+        rescue => e
```

Comment:
This is a very broad rescue, is it possible to put it around the specific lines of code that are expected to fail? 👀

Reply:
I dug deeper on this module and I think I may be able to remove this in favor of just moving the `shutdown_socket` into an `ensure` block.